app/src/test/java/com/kickstarter/viewmodels/PledgeFragmentViewModelTest.kt

```diff
@@ -137,6 +146,58 @@ class PledgeFragmentViewModelTest : KSRobolectricTestCase() {
         this.pledgeAmount.assertValue("$20")
     }
 
+    @Test
+    fun testPledgeStepping() {
+        setUpEnvironment(environment())
+
+        this.decreasePledgeButtonIsEnabled.assertValue(false)
+        this.increasePledgeButtonIsEnabled.assertValue(true)
+        this.additionalPledgeAmountIsGone.assertValue(true)
+        this.additionalPledgeAmount.assertValue("$0")
+
+        this.vm.inputs.increasePledgeButtonClicked()
+
+        this.decreasePledgeButtonIsEnabled.assertValues(false, true)
+        this.increasePledgeButtonIsEnabled.assertValue(true)
+        this.additionalPledgeAmountIsGone.assertValues(true, false)
+        this.additionalPledgeAmount.assertValues("$0", "$1")
+
+        this.vm.inputs.decreasePledgeButtonClicked()
+
+        this.decreasePledgeButtonIsEnabled.assertValues(false, true, false)
+        this.increasePledgeButtonIsEnabled.assertValue(true)
+        this.additionalPledgeAmountIsGone.assertValues(true, false, true)
+        this.additionalPledgeAmount.assertValues("$0", "$1", "$0")
+    }
+
+    @Test
+    fun testPledgeStepping_maxReward() {
+        setUpEnvironment(environment(), RewardFactory.maxReward())
+
+        this.decreasePledgeButtonIsEnabled.assertValue(false)
+        this.increasePledgeButtonIsEnabled.assertValue(false)
+        this.additionalPledgeAmountIsGone.assertValue(true)
+        this.additionalPledgeAmount.assertValue("$0")
+    }
+
+    @Test
+    fun testPledgeStepping_almostMaxReward() {
+        val oneLessThanMax = RewardUtils.MAX_REWARD_LIMIT - 1
+        setUpEnvironment(environment(), RewardFactory.reward().toBuilder().minimum(oneLessThanMax).build())
+
+        this.decreasePledgeButtonIsEnabled.assertValue(false)
```

Comment:
Why the `decreasePledgeButton` is disabled in this case?

Reply:
The button's initial state is disabled because you can't go lower than the reward minimum